docs/datastorage/capabilities.md

```diff
@@ -0,0 +1,176 @@
+
+```Java
+private static class Storage
+    implements Capability.IStorage<IExampleCapability>
+{
+  @Override
+  public NBTBase writeNBT(Capability<IExampleCapability> capability, IExampleCapability instance, EnumFacing side)
+  {
+    return instance.serializeNBT();
+  }
+
+  @Override
+  public void readNBT(Capability<IExampleCapability> capability, IExampleCapability instance, EnumFacing side, NBTBase nbt)
+  {
+    instance.deserializeNBT((NBTTagCompound)nbt);
+  }
+}
+```
+
+The last parameter is a callable factory that will return new instances of the default implementation.
+
+```Java
+private static class Factory implements Callable<IExampleCapability>
+{
+  @Override
+  public IExampleCapability call() throws Exception
+  {
+    return new Implementation();
+  }
+}
+```
+
+Finally, we will need the default implementation itself, to be able to instantiate it in the factory. Designing this class is up to you, but it should at least provide a basic skeleton that people can use to test the capability, if it's not a fully usable implementation on itself.
+
+Synchronizing Data with Clients
+-------------------------------
+
+By default, Capabilities do not synchronize. In order to change this, the mods have to manage their own synchronization code using packets.
+
+There are three different situation in which you may want to send synchronization packets, all of them optional:
+
+1. When the entity spawns in the world, or the block is placed, you may want to share the initialization-assigned values with the clients.
+1. When the stored data changes, you may want to notify some or all of the watching clients.
+1. When a new client starts viewing the entity or block, you may want to notify it of the existing data.
+
+Refer to the [Networking](../networking/index.md) page for more information on implementing the network packets.
+
+Persisting across Player Deaths
+-------------------------------
+
+By default, the capability data does not persist on death. In order to change this, the data has to be manually copied when the player entity is cloned during the respawn process.
+
+This can be done by handling the `PlayerEvent.Clone` event. In this event, the `wasDead` field can be used to distinguish between respawning after death, and returning from the End. This is important because the data will already exist, so care has to be taken to not duplicate values when returning from the End dimension.
+
+This could be done like this:
+
+```Java
+@SubscribeEvent
+public void clonePlayer(PlayerEvent.Clone e)
+{
+  if (e.wasDeath)
+  {
+    if (e.original.hasCapability(ExampleCapability.INSTANCE, null) &&
+        e.entityPlayer.hasCapability(ExampleCapability.INSTANCE, null))
+    {
+      NBTTagCompound tag = e.original.getCapability(ExampleCapability.INSTANCE, null).serializeNBT();
+      e.entityPlayer.getCapability(ExampleCapability.INSTANCE, null).deserializeNBT(tag);
+    }
+  }
+}
+```
```

Comment:
May need to write something a bit cleaner in forge to handle this but ya the whole 'lets create a new entity when respawning/changing worlds is annoying' 
However our docs SHOULD NOT encourage people to do this as it's going to create copy pasta code that will make every capability implement serializeable which is not the intention.
Probably would be best if there was a PR to make this issue act like ItemStack's clone() method, where IT called the serialize function's internally.